pkg/compact/compact.go

```diff
@@ -458,74 +385,91 @@ func newGroupMetrics(reg prometheus.Registerer, labels map[string]string, getBlo
 // Group captures a set of blocks that have the same origin labels.
 // Those blocks generally contain the same series and can thus efficiently be compacted.
 type Group struct {
-	logger  log.Logger
-	mtx     sync.Mutex
-	dir     string
-	bkt     objstore.Bucket
-	labels  map[string]string
-	metrics *groupMetrics
+	logger     log.Logger
+	bkt        objstore.Bucket
+	labels     labels.Labels
+	resolution int64
+	mtx        sync.Mutex
+	blocks     map[ulid.ULID]*block.Meta
+	metrics    *groupMetrics
 }
 
 // newGroup returns a new compaction group.
-func newGroup(logger log.Logger, reg prometheus.Registerer, bkt objstore.Bucket, dir string, labels map[string]string) (*Group, error) {
+func newGroup(
+	logger log.Logger,
+	reg prometheus.Registerer,
+	bkt objstore.Bucket,
+	lset labels.Labels,
+	resolution int64,
+) (*Group, error) {
 	if logger == nil {
 		logger = log.NewNopLogger()
 	}
-	if err := os.MkdirAll(dir, 0777); err != nil {
-		return nil, err
-	}
 	g := &Group{
-		logger: logger,
-		dir:    dir,
-		bkt:    bkt,
-		labels: labels,
+		logger:     logger,
+		bkt:        bkt,
+		labels:     lset,
+		resolution: resolution,
+		blocks:     map[ulid.ULID]*block.Meta{},
 	}
-	g.metrics = newGroupMetrics(reg, labels, g.IDs)
+	g.metrics = newGroupMetrics(reg, lset, g.IDs)
 	return g, nil
 }
 
+// Key returns an identifier for the group.
+func (cg *Group) Key() string {
+	return fmt.Sprintf("%s@%d", cg.labels, cg.resolution)
```

Comment:
hm, cannot we use the same function/form as in `groupKey` function?

Reply:
Switched form but left it as 2 functions. Just wasn't practical to consolidate.